### DISCUSSION TITLE
allow opt (mac) or ctrl (windows/linux) + backspace to delete entire word

### DISCUSSION THREAD

**Discussion by roubaobaozi** [nvlgzr/fngrng#5] [Your Carefully Crafted Thoughts]
Pressing backspace multiple times instead of just backspacing the entire word all at once is slow

**Comment by nvlgzr** [comment]
Thanks for bringing this up, @roubaobaozi! You've actually touched on a core differentiating detail in _fngrng_ which bears some explanation.

First, let me say that your request makes perfect sense to me. In fact, I was tempted during early development to implement these shortcuts. (Well, the Mac one, anyway…I haven't used Windows since 2007!) Only after I'd observed my own learning process with colemak did I decide against it. Here's why.

While I agree that it's critical for any decent text editor to support those shortcuts, my operating hypothesis is that it's actually harmful for a typing practice app to support them. If _fngrng_ let you practice using deletion shortcuts just to increase your apparent typing speed, you might find yourself developing sequences like this into muscle memory: "F-R-O-E-⌥⌫-F-O-R-E-I-G-N." For a touch-typist, those extra five characters can go by so quickly that this seems fine.

Since your maximum speed is limited by your maximum accuracy, _fngrng_ explicitly values accuracy over speed. Not only do you have to slow down enough to delete every incorrectly-typed letter, but each deletion counts as an incorrect keystroke when reporting the accuracy at the end of each session. This mild discomfort will, hopefully, encourage you to slow down enough to get your fingering right, even if it means having to be more deliberate than you're used to.

I think you'll find that your overall speed improves even _faster_ when you learn to anticipate errors before they can escape your fingertips.

Make sense?

**Reply by davidaucoin7377** [reply]
Im really a pa patient person but I'm normally on the laptop trying to learn how to do this but I have that cyclone app on a computer that I just got like today or yesterday I need some help with the dough I don't know how to connect it to a debit card maybe you can help me with that?
